### PR TITLE
✨(frontend) compact syllabus for self-paced

### LINKED ISSUE #1969: Richie course page: change order of info and simplify date presentation
## Feature Request

We believe that the course page can be improved, to prevent cases where the student don’t understand how much effort and the duration of the course. We have found cases, on support and comments on the social networks, where they simply don’t understand because they view longer start and end dates and they though that the course is very long, but in the reality the course is only a couple of hours in self pace mode. The majority of NAU courses is self pace and I think this situation is more dramatic in our case than in FUN installation, where there are a more balanced mix of self pace and instructor pace. That is why we propose the next improvement on Richie, if we don’t agree we could add it only on our installation, but if we agree on a compromise we could improve the platform for everyone.

### Restrictions
For:
- self pace course
- single open to enroll course run
- current date is in the enrollment period - course in open for enroll

Then on the course run block:
- do not show any enrollment dates.
- show only the end date - on a compact style.

### Language

I think another problem is the language per course run. It is a good feature to have different runs using different languages, but in reality almost all course runs for the course has the same language(s).
What we propose is if all course runs for the course has the same language(s) then we should present that information only once. For an example where a run is almost finish and another is starting up.
One location for this single language presentation could be after the duration, effort and pace; but before the enrollment count.
So, the course languages should be an union of all course run languages.

### Change order of Catch phrase, duration, effort and pace

If we compare Richie with other Course catalog systems, like [edx.org](http://edx.org/) or Coursera, we see that the catch phrase is right below the course title. Only after the catch phrase, it has more information, for example it shows the course duration, effort and pace. We believe that we should change the order of the information to be similar to those platforms.

We propose that we change the order of those elements in Richie. To be like: course title, catch phrase, duration, effort, pace, languages and enrollment count. Maintaining the duration, effort and pace on the same line.

<Course Title>
<catch phrase>
<duration> <effort> <pace>
<languages>
<enrollment count>

### Screen captures and Mockups

Current course presentation from the demo site:
image

(Mockup) The same demo course presented with the suggestion changes:
course_page

On NAU this change would have this effect:
nau_course_card

**Describe alternatives you've considered**
Implement part on NAU Richie site factory

**Do you want to work on it through a Pull Request?**
Yes.

### PR DESCRIPTION
**Pull request by JoaoGarcao**
1. If all runs of an opened course have the same languages, we hide the languages on the course run card, because those languages are shown after the course description.
2. On self-paced courses with single opened run, display a simplified version of the syllabus.

fix #1969

Change the message of the enrollment period for a course with single opened run when it is self-paced, and
hide the languages of a course when every run of that course has the same languages as all the others.

## Opened self-paced course:
This PR shows a compacted version of the information.
Before:

After:

## Opened course with the same languages as all the other runs of that course
As all runs for the course as the same languages, showing the same exact information multiple times is unnecessary, so this PR removes that languages on each run syllabus.
This behavior is the same for both self-paced and instructor-paced courses.
Before:

After:

## Instructor course with different languages, stays the same
If a course isn't self-paced (instructor pace) and the languages of all runs of that course aren't equal, then the course run card stays the same as it was:

## Compacted version for self-paced & same languages for all runs
If an opened course run is self-paced and the languages of all runs of that course are equal, then the course run card will be as displayed bellow.
This is the normal/majority of the courses in [NAU](https://www.nau.edu.pt/).

## Compacted version for self-paced, same languages for all runs & no course end date
if we have the case described above, but also there is no course end date,  then the course run card will be:

FYI @igobranco @sandroscosta

### CODE REVIEW

**Comment by igobranco** [comment]
@JoaoGarcao please fix the lint's.

```bash
make lint-back
make lint-front
```

On the last image scenario, _Compacted version for self-paced & same languages for all runs_, please remove the _Enrollment_.
This means that the header with the _ Enrollment_ as text shouldn't be displayed on the compacted version.

**Comment by igobranco** [comment]
@jbpenrath - I haven't present to you @JoaoGarcao , but he is a student that have been doing a curricular internship on NAU.
I've already reviewed this PR with him, but I think you or someone from FUN should review this PR, before we merge it.
This PR has impact on almost 100% of NAU course pages, but on you side as lower impact ( < 50% opened are self paced).

File: src/frontend/js/types/Joanie.ts
Comment (jbpenrath) [inline src/frontend/js/types/Joanie.ts:213]: Until this PR, the `CourseLight` type can be shared. But since you add the field `is_self_paced` you have to create a dedicated type.

Indeed, We are using `CourseLight` type to type Joanie API Responses but from Joanie the response will never return `is_self_paced` field.

IMO you should create a Course type within `types/index.ts`

File: CHANGELOG.md
Comment (JoaoGarcao) [reply CHANGELOG.md]: Change reverted.

File: src/frontend/js/types/Joanie.ts
Comment (JoaoGarcao) [reply src/frontend/js/types/Joanie.ts]: As you said, i created a Course type (and named it as `PacedCourse`) within `types/indes.ts`. It has a `courseLight` and the boolean variable `is_self_paced`.

File: src/frontend/js/utils/CourseRunLanguages/index.ts
Comment (JoaoGarcao) [reply src/frontend/js/utils/CourseRunLanguages/index.ts]: I looked at `CertificateHelper` and tried to do a similar thing, hopefully it looks good to you.

File: src/frontend/js/widgets/SyllabusCourseRunsList/components/SyllabusAsideList/index.tsx
Comment (JoaoGarcao) [reply src/frontend/js/widgets/SyllabusCourseRunsList/components/SyllabusAsideList/index.tsx:109]: Change done.

File: src/frontend/js/utils/CourseRunHelper/index.ts
Comment (JoaoGarcao) [reply src/frontend/js/utils/CourseRunHelper/index.ts:1]: Tests made.

File: src/frontend/js/widgets/SyllabusCourseRunsList/components/SyllabusCourseRunCompacted/index.tsx
Comment (JoaoGarcao) [reply src/frontend/js/widgets/SyllabusCourseRunsList/components/SyllabusCourseRunCompacted/index.tsx:33]: Replacement done.

File: src/frontend/js/types/index.ts
Comment (JoaoGarcao) [reply src/frontend/js/types/index.ts:23]: I did that change as you said, but i also changed the type of some variables, because there were fields that ```PacedCourse``` couldn't have.

File: src/frontend/js/utils/CourseRunHelper/index.spec.ts
Comment (jbpenrath) [inline src/frontend/js/utils/CourseRunHelper/index.spec.ts:6]:
```suggestion
  it('should return true if all course runs have the same languages', () => {
```

File: src/frontend/js/utils/CourseRunHelper/index.spec.ts
Comment (jbpenrath) [inline src/frontend/js/utils/CourseRunHelper/index.spec.ts:21]:
```suggestion
  it('should return false if all course runs have different languages', () => {
```

File: src/frontend/js/widgets/SyllabusCourseRunsList/index.tsx
Comment (jbpenrath) [inline src/frontend/js/widgets/SyllabusCourseRunsList/index.tsx:46]:
```suggestion
  course: PacedCourse;
```

With your changes, this component should now accept only `PacedCourse`

File: src/frontend/js/widgets/SyllabusCourseRunsList/index.tsx
Comment (jbpenrath) [inline src/frontend/js/widgets/SyllabusCourseRunsList/index.tsx:89]: As now course is a CoursePaced, you are able to remove this first check

**Review by jbpenrath — APPROVED** [review]
Last comments then looks good to me ✅👏

Just an overall comment for your next PRs try to split your work in smaller commits. E.g: IMO this PR could have been split into 3 commits.
1. Commit which udpate the extra tags to self_paced property in the backend application
2. Commit focused on the showLanguages feature
3. Commit focused on the new component for self paced course runs.

File: src/frontend/js/utils/CourseRunHelper/index.spec.ts
Comment (JoaoGarcao) [reply src/frontend/js/utils/CourseRunHelper/index.spec.ts]: Change done.

File: src/frontend/js/utils/CourseRunHelper/index.spec.ts
Comment (JoaoGarcao) [reply src/frontend/js/utils/CourseRunHelper/index.spec.ts]: Change done.

File: src/frontend/js/widgets/SyllabusCourseRunsList/index.tsx
Comment (JoaoGarcao) [reply src/frontend/js/widgets/SyllabusCourseRunsList/index.tsx]: I've made some changes according to what I've understood.

File: src/frontend/js/types/Joanie.ts
Comment (JoaoGarcao) [reply src/frontend/js/types/Joanie.ts]: Change done.

Comment (jbpenrath) [reply src/frontend/js/types/Joanie.ts]: Just for information, click on the ♻️ icon to notify me that you are waiting for a new review.
image

Comment (JoaoGarcao) [reply src/frontend/js/types/Joanie.ts]: Thank you, i'll start using it for now on.

File: src/frontend/js/utils/test/factories/joanie.ts
Comment (jbpenrath) [inline src/frontend/js/utils/test/factories/joanie.ts:287]: Can you just move this factory into `js/utils/test/factories/richie.ts` please.

File: src/frontend/js/types/Joanie.ts
Comment (jbpenrath) [inline src/frontend/js/types/Joanie.ts:185]:
```suggestion
```

**Review by jbpenrath — APPROVED** [review]
Last minor comment then looks good to merge 👏

**Review by igobranco — APPROVED** [review]

**Pull request merged**
